### ISSUE TITLE
no-inline-comments exception for istanbul ignore comments

### ISSUE THREAD

**Issue by coreyfarrell** [eslint/eslint#12755]
**What rule do you want to change?**
no-inline-comments

**Does this change cause the rule to produce more or fewer warnings?**
Fewer

**How will the change be implemented? (New option, new default behavior, etc.)?**
Either a new option or default behavior could work depending on what eslint maintainers think is appropriate.

**Please provide some example code that this change will affect:**

```js
if (condition1) {
  console.log('condition1');
} /* istanbul ignore else */ else if (condition2) {
  console.log('condition2');
} else {
  console.log('unreachable');
}
```

**What does the rule currently do for this code?**
Report an error

**What will the rule do after it's changed?**
Allow an exception for `istanbul ignore` comments.  This is needed as the location of this comment matters.  For it to ignore the chained else branch the comment needs to be directly before `else if` or between the `else` and `if`.

**Are you willing to submit a pull request to implement this change?**
I can try if this is an acceptable change and if I'm given guidance about an option vs default behavior.

**Comment by ilyavolodin** [comment]
Can you provide an error that you are seeing?

**Comment by coreyfarrell** [comment]
eslint output:
```
/usr/src/npm/tmp/test.js
  3:3  error  Unexpected comment inline with code  no-inline-comments

✖ 1 problem (1 error, 0 warnings)
```

**Comment by mdjermanovic** [comment]
An option where user can define a regex pattern (or array of patterns?) to ignore comments that match the pattern seems reasonable to me.

**Comment by kaicataldo** [comment]
I think an option to allow this would be fine. Alternatively, you can disable the `no-inline-comments` rule around these comments using `eslint-disable` comments.

**Comment by mdjermanovic** [comment]
I think it would be useful to support this in some way :+1:  

Exceptions (default and options) are already supported in other comment rules: [line-comment-position](https://eslint.org/docs/rules/line-comment-position), [capitalized-comments](https://eslint.org/docs/rules/capitalized-comments) and [lines-around-comment](https://eslint.org/docs/rules/lines-around-comment).

`no-inline-comments` is very similar to `line-comment-position` (actually, it looks like they're overlapping/conflicting for line comments?).

**Comment by mdjermanovic** [comment]
I'll champion this.

**Comment by mdjermanovic** [comment]
This will be fixed by #13029, we'll add `"ignorePattern"` option (accepts `string` that will be treated as regex) which will allow users to configure this rule to ignore `istanbul` comments.
